pandas/core/internals/array_manager.py

```diff
@@ -729,6 +737,10 @@ def fast_xs(self, loc: int) -> ArrayLike:
         result = np.array([arr[loc] for arr in self.arrays], dtype=temp_dtype)
         if isinstance(dtype, ExtensionDtype):
             result = dtype.construct_array_type()._from_sequence(result, dtype=dtype)
+        elif is_datetime64_ns_dtype(dtype):
+            result = DatetimeArray._from_sequence(result, dtype=dtype)._data
+        elif is_timedelta64_ns_dtype(dtype):
+            result = TimedeltaArray._from_sequence(result, dtype=dtype)._data
```

Comment:
The reason I am doing this here is because this fails:

```
In [4]: np.array([pd.NaT], dtype="M8[ns]")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-99d42e913a1c> in <module>
----> 1 np.array([pd.NaT], dtype="M8[ns]")

ValueError: cannot convert float NaN to integer
```

(which is what the `np.array([arr[loc] for arr in self.arrays], dtype=temp_dtype)` above can be doing if the resulting dtype is `M/m8`)

@jbrockmendel do you know if that's something we do elsewhere as well / there is some existing code for this?

Reply:
IIRC we can get BM to mess up if we get here with non-consolidated all-td64 blocks.

I think the thing to do (also similar change in the BM method) is to define result on L737 as a list, only wrap with np.array if none of these conditions hold

Reply:
> IIRC we can get BM to mess up if we get here with non-consolidated all-td64 blocks.

You can't get here in that case, though, since this is an ArrayManager method. And with the current BM you don't have this problem since it doesn't store DatetimeArray/TimedeltaArray with numpy dtypes. 

> I think the thing to do (also similar change in the BM method) is to define result on L737 as a list, only wrap with np.array if none of these conditions hold

Thanks, that's a good idea

Reply:
Updated with this simplification.